src/test/java/gov/usgs/owi/nldi/controllers/LinkedDataControllerTest.java

```diff
@@ -78,35 +80,47 @@ public void getComidTest() {
 		assertNull(controller.getComid("abc", "def"));
 	}
 
+	@Test(expected = IllegalArgumentException.class)
+	public void getComidWithNullFeatureSourceTest() {
+		controller.getComid(null, "def");
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void getComidWithNullFeatureIdTest() {
+		controller.getComid("FakeFeatureSource", null);
+	}
+
 	@Test
 	@SuppressWarnings("unchecked")
 	public void getFlowlinesTest() throws Exception {
 		when(lookupDao.getComid(anyString(), anyMap())).thenReturn(null, goodFeature());
-		controller.getFlowlines(request, response, null, null, null, null, null, null);
+		controller.getFlowlines(request, response, "DoesntMatter", "DoesntMatter", null, null, null, null);
 		verify(logService).logRequest(any(HttpServletRequest.class));
 		verify(logService).logRequestComplete(any(BigInteger.class), any(int.class));
+		//Mock lookupDao 1st response of null means the comid is not found, thus a 404
 		assertEquals(HttpStatus.NOT_FOUND.value(), response.getStatus());
 
 		controller.getFlowlines(request, response, null, null, null, null, null, null);
 		verify(logService, times(2)).logRequest(any(HttpServletRequest.class));
 		verify(logService, times(2)).logRequestComplete(any(BigInteger.class), any(int.class));
-		//this is a INTERNAL_SERVER_ERROR because of NPEs that shouldn't happen in real life.
+		//Mock lookupDao 2nd response doesn't actually exist, thus causes a 500 when we try to get flowlines
 		assertEquals(HttpStatus.INTERNAL_SERVER_ERROR.value(), response.getStatus());
```

Comment:
This second part of this test (and others like it) is really confusing because we are artificially creating the INTERNAL_SERVER_ERROR as a side-effect of mocking the lookupDao to return a non-existing id.  Really it just tests how the system handles the mocks since the actual two requests are identical.

Best world:  It would be nice to clean up these tests by adding real lookup data for these tests to hit.

Reply:
This entire class is unit testing the controller. The second response is showing that we have captured the unanticipated error, converting it to an INTERNAL_SERVER_ERROR (500) and thus not leaking implementation details to the outside world.